src/waiting_call_registry.rs

```diff
@@ -18,11 +18,17 @@ impl IdentifiableResponse for Response {
     }
 }
 
-impl WaitingCallRegistry {
-    pub fn new() -> Self {
+impl Default for WaitingCallRegistry {
```

Comment:
Similarly here, what's the advantage? (I'm fairly new to Rust!)

Reply:
Not sure if there's a real advantage. It's just something that clippy suggested. I guess it makes sense. If the `new` function doesn't get any parameters then it's basically like saying that the struct implements `Default`